src/classes/ADDR_CopyAddrHHObjBTN_TEST.cls

```diff
@@ -33,84 +33,295 @@
+    * @description Populates the record variables with the records generated in test setup
+    * @return void
+    */
+    private static void getRecordData() {
+        householdRecord = getHouseholdRecordData();
+        contactRecord = getContactRecordData();
+    }
+
+    /*******************************************************************************************************************
+    * @description Queries the household record created in test setup
+    * @return npo02__Household__c
+    */
+    private static npo02__Household__c getHouseholdRecordData() {
+        return [
+            SELECT Id, npo02__MailingStreet__c, npo02__MailingState__c,
+                npo02__MailingPostalCode__c, npo02__MailingCity__c, npo02__MailingCountry__c
+            FROM npo02__Household__c
+            LIMIT 1
+        ];
+    }
+
+    /*******************************************************************************************************************
+    * @description Queries the contact record created in test setup
+    * @return Contact
+    */
+    private static Contact getContactRecordData() {
+        return [
+            SELECT Id, MailingStreet, MailingState, MailingPostalCode, MailingCity, MailingCountry
+            FROM Contact
+            LIMIT 1
+        ];
+    }
+
+    /*******************************************************************************************************************
+    * @description Sets the current page to page controller from the button click of provided record Id
+    * with user having permissions by default.
+    * @param recordId - the record Id to pass to the controller
+    */
+    private static void createCopyAddressBtnCtrl(Id recordId) {
         Test.setCurrentPageReference(new PageReference('Page.ADDR_CopyAddrHHObjBTN'));
-        system.currentPageReference().getParameters().put('id', null);
-        cacbc = new ADDR_CopyAddrHHObjBTN_CTRL();
-        prc = cacbc.buttonClick();
-        system.assert(prc == null);
+        System.currentPageReference().getParameters().put('id', recordId);
+        copyAddressCtrl = new ADDR_CopyAddrHHObjBTN_CTRL();
+        copyAddressCtrl.perms = (UTIL_Permissions) Test.createStub(UTIL_Permissions.class, mockPerms);
+
+        copyAddressCtrl.buttonClick();
+    }
+
+    /*******************************************************************************************************************
+    * @description Verify a note was created for the provided record Id
+    * @param parentId - the id of the record the note belongs to
+    * @return void
+    */
+    private static void verifyNoteRecordCount(Id parentId) {
+        System.assertEquals(1, [SELECT count() FROM Note where ParentID = :parentId]);
     }
+
+    /*******************************************************************************************************************
+    * @description Stub for SObjectDescribe instance
```

Comment:
Looks like the description from needs to updated from SObjectDescribe to PermissionsMock